app/src/Application.php

```diff
@@ -130,4 +118,34 @@ protected function bootstrapCli(): void
 
         // Load more plugins here
     }
+
+
+    public function getAuthenticationService(ServerRequestInterface $request): AuthenticationServiceInterface
+    {
+        $authenticationService = new AuthenticationService([
+            'unauthenticatedRedirect' => Router::url('/users/login'),
```

Comment:
```suggestion
            'unauthenticatedRedirect' => Router::url(['controller' => 'Users', 'action' => 'login']),
```

Instead of using string URL, we normally used the array format routing. It's more verbose and easy to maintain later (The URL string might changed but the controller and action rarely changed).

Please refer https://book.cakephp.org/4/en/development/routing.html#routing-arrays-vs-request-parameters

Reply:
Please change to the routing array approach as mentioned above.

Reply:
```suggestion
            'unauthenticatedRedirect' => Router::url(['controller' => 'Users', 'action' => 'login']),
```

It's fixed in https://github.com/ArielGalvez/tutorial-cakephp/blob/f4782e36bc41c862e369ba52fbd781c86fa0fe17/app/src/Application.php#L172

However, this line remains not fix. Please check again.

Reply:
now is fixed